src/compile/import_resolver.ts

```diff
@@ -8,27 +8,55 @@ export interface ImportResolver {
     resolve(fileName: string): string | undefined;
 }
 
+export type Remapping = [string, string, string];
+
 export class FileSystemResolver implements ImportResolver {
+    basePath?: string;
+    includePaths?: string[];
+
+    constructor(basePath?: string, includePaths?: string[]) {
+        this.basePath = basePath;
+        this.includePaths = includePaths;
+    }
+
     resolve(fileName: string): string | undefined {
-        return fse.existsSync(fileName) ? fileName : undefined;
+        const prefixes = [this.basePath ? this.basePath : ""].concat(
+            this.includePaths ? this.includePaths : []
+        );
+
+        for (const prefix of prefixes) {
+            let candidate: string;
+
+            if (prefix) {
+                const relative = path.relative(prefix, fileName);
+
+                candidate = path.join(prefix, relative.startsWith("../") ? fileName : relative);
```

Comment:
We try to figure out if cimpiler input source file path is relative to base path. In that case we want to avoid double prefix of base path. Relative path would contain `../` in case if they are not relative. Consider looking [here](https://github.com/ethereum/solc-js/blob/99eafc2ad13d5bfba7886de659a6638814d95078/solc.ts#L108).